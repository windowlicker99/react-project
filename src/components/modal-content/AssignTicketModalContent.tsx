import React, { ChangeEventHandler, FC } from 'react';
import { FormikValues } from 'formik';
import { Textarea } from '@components/common/form-controls/Textarea';
import { SelectComponent } from '@components/common/form-controls/SelectComponent';
import { IOption } from '@interfaces/interfaces';

interface IAssignTicketModalContent {
  values: FormikValues;
  options: IOption[];
  onChange: ChangeEventHandler;
  onSelectChange: (name: string, value: IOption) => void;
}

export const AssignTicketModalContent: FC<IAssignTicketModalContent> = ({
  values,
  options,
  onChange,
  onSelectChange,
}) => (
  <div className="modal-info">
    <SelectComponent
      name="user"
      value={values.user}
      options={options}
      label="Assign to"
      placeholder=""
      onChange={onSelectChange}
    />
    <Textarea placeholder="Add a message (optional)" name="message" value={values.message} onChange={onChange} />
  </div>
);
