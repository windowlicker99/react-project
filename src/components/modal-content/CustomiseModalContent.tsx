import React, { ChangeEventHandler, FC, FormEventHandler } from 'react';
import { FormikValues } from 'formik';
import { Checkbox } from '@components/common/form-controls/Checkbox';

interface IColumnField {
  label: string;
  name: string;
}

interface ICustomiseModalContent {
  handleSubmit: FormEventHandler<HTMLFormElement>;
  handleChange: ChangeEventHandler<HTMLInputElement>;
  values: FormikValues;
  columns: IColumnField[];
}

export const CustomiseModalContent: FC<ICustomiseModalContent> = ({ values, columns, handleChange, handleSubmit }) => (
  <div className="modal-info">
    <form onSubmit={handleSubmit}>
      <ul className="customise-list">
        {columns.map(({ label, name }) => (
          <li className="customise-item" key={name}>
            <button type="button" className="customise-move" />
            <Checkbox name={name} checked={values[name]} onChange={handleChange} label={label} />
          </li>
        ))}
      </ul>
    </form>
  </div>
);
