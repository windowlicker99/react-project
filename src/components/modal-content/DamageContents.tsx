import { FormikValues, useFormikContext } from 'formik';
import React, { FC } from 'react';
import { SelectComponent } from '@components/common/form-controls/SelectComponent';
import { setDamagePotential } from '@/constants/formsElements';
import { deleteInitialValues } from '@/utils/deleteInitialValues';

const title = [{ name: 'Confidentiality' }, { name: 'Integrity' }, { name: 'Availability' }, { name: 'Authenticity' }];

const formInputs = deleteInitialValues(setDamagePotential);

export const DamageContent: FC = () => {
  const { values, setFieldValue } = useFormikContext<FormikValues>();
  return (
    <div className="damage-content">
      {title.map(({ name }) => (
        <React.Fragment key={name}>
          <span className="damage-title">{name}</span>
          <div className="damage-item">
            {Object.keys(formInputs).map((input) => (
              <SelectComponent
                key={input}
                onChange={setFieldValue}
                {...formInputs[input]}
                value={values[name][input]}
                name={`${name}.${input}`}
              />
            ))}
          </div>
        </React.Fragment>
      ))}
    </div>
  );
};
