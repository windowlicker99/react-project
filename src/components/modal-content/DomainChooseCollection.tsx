// markup only
import React, { FC } from 'react';
import { CardDropdown } from '@components/common/Ð¡ardDropdown';
import { ModalControl } from '@components/common/ModalControl';

export const DomainChooseContent: FC = () => (
  <div className="domains-aside-item">
    <ModalControl />
    <CardDropdown name="domain" />
  </div>
);
