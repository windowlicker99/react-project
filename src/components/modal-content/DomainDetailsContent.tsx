import React, { FC } from 'react';
import { FormikProps, FormikValues } from 'formik';
import { Input } from '@components/common/form-controls/Input';
import { BadgeRoundGroup } from '@components/common/BadgeRoundGroup';
import { SelectComponent } from '@components/common/form-controls/SelectComponent';
import { FormInputSwitch } from '@components/common/form-controls/FormInputSwitch';
import { IOption } from '@interfaces/interfaces';
import { domainCreationElements } from '@constants/formsElements';
import { deleteInitialValues } from '@utils/deleteInitialValues';

interface IDomainDetailsContent {
  handleSelectChange: (name: string, value: IOption) => void;
}

export const DomainDetailsContent: FC<IDomainDetailsContent & Partial<FormikProps<FormikValues>>> = ({
  values,
  errors,
  touched,
  handleBlur,
  handleChange,
  handleSelectChange,
}) => {
  const { name, usage, connectivity, communication, color } = deleteInitialValues(domainCreationElements);

  return (
    <div className="domains-aside-item">
      <div className="modal-box">
        <Input
          {...name}
          error={touched[name.name] && (errors[name.name] as string)}
          onChange={handleChange}
          value={values[name.name]}
          onBlur={handleBlur}
        />
        <Input
          {...usage}
          error={touched[usage.name] && (errors[usage.name] as string)}
          onChange={handleChange}
          value={values[usage.name]}
          onBlur={handleBlur}
        />
        <FormInputSwitch value={values[connectivity.name]} {...connectivity} onChange={handleChange} />
        <SelectComponent
          error={touched[communication.name] && (errors[communication.name] as string)}
          value={values[communication.name]}
          {...communication}
          onChange={handleSelectChange}
        />
      </div>
      <BadgeRoundGroup value={values[color.name]} {...color} onChange={handleChange} />
    </div>
  );
};
