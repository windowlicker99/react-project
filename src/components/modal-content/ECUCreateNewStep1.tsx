import React, { FC, useState, useEffect } from 'react';
import { Field, FormikValues, useFormikContext } from 'formik';
import { SelectComponent } from '@components/common/form-controls/SelectComponent';
import { SelectInfo } from '@components/common/form-controls/SelectInfo';
import { FormikSelectsArrayField } from '@components/common/form-controls/FormikSelectsArrayField';
import { deleteInitialValues } from '@utils/deleteInitialValues';
import { getSuppliers } from '@utils/getSuppliers';
import { convertsResponseDataToOptions } from '@utils/converters';
import { getDomains, getECUs } from '@utils/getElements';
import { ECUCreation } from '@constants/formsElements';
import { IOption } from '@interfaces/interfaces';

interface IECUCreateNewStep1 {
  vehicleId: string;
  updateContent: boolean;
}

export const ECUCreateNewStep1: FC<IECUCreateNewStep1> = ({ vehicleId, updateContent }) => {
  const [supplierOptions, setSupplierOptions] = useState<IOption[]>([]);
  const [ECUsOptions, setECUsOptions] = useState<IOption[]>([]);
  const [hostedDomainsOptions, setHostedDomainsOptions] = useState<IOption[]>([]);

  const { values, initialValues, errors, touched, setFieldValue } = useFormikContext<FormikValues>();

  const handleSelectChange = (name: string, value: IOption) => setFieldValue(name, value);

  const {
    name,
    operationSystem,
    connectivity,
    firmwareVersion,
    supplier,
    isRegular,
    isGateway,
    isSwitch,
    hostVMorDocker,
    runVMorDocker,
    hostECU,
    domains,
  } = deleteInitialValues(ECUCreation);

  const baseECUInfoFields = [name, operationSystem, connectivity, firmwareVersion];

  const switches = [isRegular, isGateway, isSwitch];

  const switchesForIrregularECU = [hostVMorDocker, runVMorDocker];

  const getSuppliersOptions = async () => {
    const data = await getSuppliers();
    const options = convertsResponseDataToOptions(data);
    setSupplierOptions(options);
  };

  const getDomainsOptions = async () => {
    if (!vehicleId) {
      setHostedDomainsOptions([]);
      return;
    }
    const data = await getDomains(vehicleId);
    const options = convertsResponseDataToOptions(data);
    setHostedDomainsOptions(options);
  };

  const getECUsOptions = async () => {
    if (!vehicleId) {
      setECUsOptions([]);
      return;
    }
    const data = await getECUs(vehicleId);
    const options = convertsResponseDataToOptions(data);
    setECUsOptions(options);
  };

  const onIsECURegularChange = () =>
    switchesForIrregularECU.forEach(({ name }) => setFieldValue(name, initialValues[name]));

  useEffect(() => {
    getSuppliersOptions();
  }, []);

  useEffect(() => {
    getDomainsOptions();
    getECUsOptions();
  }, [vehicleId, updateContent]);

  useEffect(() => {
    if (values[isRegular.name]) {
      onIsECURegularChange();
    }
  }, [values[isRegular.name]]);

  return (
    <div className="ecu-details-content">
      <div className="grid-box-columns">
        {baseECUInfoFields.map(({ name, ...field }) => (
          <Field key={name} error={touched[name] && errors[name]} name={name} {...field} />
        ))}
        <div>
          <SelectComponent
            value={values[supplier.name]}
            options={supplierOptions}
            {...supplier}
            onChange={handleSelectChange}
          />
          <div style={{ color: 'red', fontSize: '14px' }}>{errors[supplier.name]}</div>
        </div>
      </div>
      <div className="card-dropdown">
        {switches.map(({ name, ...field }) => (
          <Field name={name} {...field} key={name} />
        ))}
        {!values[isRegular.name] && (
          <>
            {switchesForIrregularECU.map(({ name, ...field }) => (
              <Field name={name} {...field} key={name} />
            ))}
            <SelectInfo {...hostECU} value={values[hostECU.name]} options={ECUsOptions} onChange={handleSelectChange} />
          </>
        )}
      </div>
      <FormikSelectsArrayField
        field={domains}
        selects={Object.values(domains.subFields)}
        asyncOptions={hostedDomainsOptions}
        handleSelectChange={handleSelectChange}
        btnText="Add Hosted Domain"
      />
    </div>
  );
};
