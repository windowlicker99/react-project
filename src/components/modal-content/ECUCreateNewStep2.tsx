import React, { FC, useState, useEffect } from 'react';
import { Field, FormikValues, useFormikContext } from 'formik';
import { SelectComponent } from '@components/common/form-controls/SelectComponent';
import { FormikSelectsArrayField } from '@components/common/form-controls/FormikSelectsArrayField';
import { deleteInitialValues } from '@utils/deleteInitialValues';
import { convertsResponseDataToOptions } from '@utils/converters';
import { getECUs } from '@utils/getElements';
import { ECUCreation } from '@constants/formsElements';
import { IOption } from '@interfaces/interfaces';

interface IECUCreateNewStep2 {
  vehicleId: string;
  updateContent: boolean;
}

export const ECUCreateNewStep2: FC<IECUCreateNewStep2> = ({ vehicleId, updateContent }) => {
  const [ECUsOptions, setECUsOptions] = useState<IOption[]>([]);

  const { values, initialValues, setFieldValue } = useFormikContext<FormikValues>();

  const handleSelectChange = (name: string, value: IOption | IOption[]) => {
    setFieldValue(name, value);
  };

  const {
    haveDirectCommunication,
    directCommunication,
    updateProcess,
    managingOrUsing,
    storeConfidentialData,
    useCryptographicTechnologies,
    indirectCommunicationChannel,
    connectedECUs,
  } = deleteInitialValues(ECUCreation);

  const resetField = (name: string) => {
    setFieldValue(name, initialValues[name]);
  };

  const getECUsOptions = async () => {
    if (!vehicleId) {
      setECUsOptions([]);
      return;
    }
    const data = await getECUs(vehicleId);
    const options = convertsResponseDataToOptions(data);
    setECUsOptions(options);
  };

  useEffect(() => {
    getECUsOptions();
  }, [vehicleId, updateContent]);

  useEffect(() => {
    if (!values[haveDirectCommunication.name]) {
      resetField(directCommunication.name);
    }
  }, [values[haveDirectCommunication.name]]);

  useEffect(() => {
    if (!values[indirectCommunicationChannel.name]) {
      resetField(connectedECUs.name);
    }
  }, [values[indirectCommunicationChannel.name]]);

  useEffect(() => {
    if (!values[updateProcess.name]) {
      resetField(managingOrUsing.name);
    }
  }, [values[updateProcess.name]]);

  return (
    <div className="ecu-details-content">
      <div className="card-dropdown">
        <div className="card-dropdown-item">
          <Field {...haveDirectCommunication} />
          {values[haveDirectCommunication.name] && (
            <SelectComponent
              {...directCommunication}
              value={values[directCommunication.name]}
              onChange={handleSelectChange}
            />
          )}
        </div>
        <div className="card-dropdown-item">
          <Field {...updateProcess} />
          {values[updateProcess.name] &&
            managingOrUsing.values.map((value) => (
              <Field key={value} name={managingOrUsing.name} label={value} component={managingOrUsing.component} />
            ))}
        </div>

        <Field {...storeConfidentialData} />
        <Field {...useCryptographicTechnologies} />
        <Field {...indirectCommunicationChannel} />
      </div>

      {values[indirectCommunicationChannel.name] && (
        <FormikSelectsArrayField
          field={connectedECUs}
          selects={Object.values(connectedECUs.subFields)}
          asyncOptions={ECUsOptions}
          handleSelectChange={handleSelectChange}
          btnText="Add ECU"
        />
      )}
    </div>
  );
};
