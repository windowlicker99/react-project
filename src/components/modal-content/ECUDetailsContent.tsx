import React, { FC } from 'react';
import { Btn } from '../common/form-controls/Btn';
import { FormInputSwitch } from '../common/form-controls/FormInputSwitch';
import { Input } from '../common/form-controls/Input';
import { SelectComponent } from '../common/form-controls/SelectComponent';
import { SelectInfo } from '../common/form-controls/SelectInfo';
import { SwitchInfo } from '../common/form-controls/SwitchInfo';
import { Upload } from '../common/form-controls/Upload';

export const ECUDetailsContent: FC = () => (
  <div className="ecu-details-content">
    <div className="grid-box-columns">
      <Input name="test" label="label" placeholder="placeholder" />
      <SelectComponent label="Domain Usage" placeholder="Communication" />
      <FormInputSwitch />
      <Input name="test" label="label" placeholder="placeholder" />
      <SelectComponent label="Domain Usage" placeholder="Communication" />
    </div>
    <div className="card-dropdown">
      <SwitchInfo label="Lorem ipsum" />
      <SwitchInfo label="Lorem ipsum" />
      <SwitchInfo label="Lorem ipsum" />
      <SelectInfo label="Lorem ipsum" />
    </div>
    <div className="grid-box-columns-btn">
      <SelectComponent label="Domain Usage" placeholder="Communication" />
      <SelectComponent label="Domain Usage" placeholder="Communication" />
      <Btn btnStatus="btn-xs-danger" btnIcon="icon-delete" />
    </div>
    <div className="grid-box-columns-btn">
      <SelectComponent label="Domain Usage" placeholder="Communication" />
      <SelectComponent label="Domain Usage" placeholder="Communication" />
      <Btn btnStatus="btn-xs-danger" btnIcon="icon-delete" />
    </div>
    <Upload />
    <Btn btnStatus="btn-lg-outline" btnIcon="" btnText="Add Data" />
  </div>
);
