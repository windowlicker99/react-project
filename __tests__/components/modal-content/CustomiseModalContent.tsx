import React from 'react';
import { render } from '@testing-library/react';
import { CustomiseModalContent } from '@components/modal-content/CustomiseModalContent';

describe('CustomiseModalContent tests', () => {
  const columns = [
    { label: 'Name', name: 'name' },
    { label: 'Date', name: 'date' },
  ];

  const formik = {
    values: { date: false, name: true },
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
  };

  it('Should render', () => {
    const { container } = render(<CustomiseModalContent columns={columns} {...formik} />);
    expect(container.getElementsByClassName('customise-list').length).toEqual(1);
  });

  it('Should match snapshot', () => {
    const { asFragment } = render(<CustomiseModalContent columns={columns} {...formik} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
