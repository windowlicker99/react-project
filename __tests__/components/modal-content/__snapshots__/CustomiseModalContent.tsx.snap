// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomiseModalContent tests Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal-info"
  >
    <form>
      <ul
        class="customise-list"
      >
        <li
          class="customise-item"
        >
          <button
            class="customise-move"
            type="button"
          />
          <div
            class="checkbox"
          >
            <label
              class="checkbox-label"
              for="name"
            >
              <input
                checked=""
                class="checkbox-input"
                id="name"
                name="name"
                type="checkbox"
              />
              <span
                class="checkbox-mark"
              />
              <span
                class="checkbox-title"
              >
                Name
              </span>
            </label>
          </div>
        </li>
        <li
          class="customise-item"
        >
          <button
            class="customise-move"
            type="button"
          />
          <div
            class="checkbox"
          >
            <label
              class="checkbox-label"
              for="date"
            >
              <input
                class="checkbox-input"
                id="date"
                name="date"
                type="checkbox"
              />
              <span
                class="checkbox-mark"
              />
              <span
                class="checkbox-title"
              >
                Date
              </span>
            </label>
          </div>
        </li>
      </ul>
    </form>
  </div>
</DocumentFragment>
`;
